iac-extensions/docker/src/test/java/org/sonar/iac/docker/reports/HadoLintImporterTest.java

```diff
@@ -0,0 +1,177 @@
+    "src/test/resources/hado-lint/parseError.json, Hado-lint report importing: could not parse file as JSON %s",
+    "src/test/resources/hado-lint/noArray.json, Hado-lint report importing: file is expected to contain a JSON array but didn't %s"})
+  void problem_when_reading_or_parsing_file(String reportPath, String expectedLog) {
+    String path = File.separatorChar == '/' ? reportPath : Paths.get(reportPath).toString();
+    File reportFile = new File(path);
+    String logMessage = String.format(expectedLog, path);
+
+    importReport(reportFile);
+    assertThat(logTester.logs(LoggerLevel.WARN)).containsExactly(logMessage);
+    verify(mockAnalysisWarnings, times(1)).addWarning(logMessage);
+  }
+
+  @Test
+  void reading_issue() {
+    String path = "src\\test\\resources\\hado-lint\\throwsIOException.json";
+    File reportFile = Mockito.mock(File.class);
+    String logMessage = String.format("Hado-lint report importing: could not read report file %s", path);
+    when(reportFile.getPath()).thenReturn(path);
+    when(reportFile.isFile()).thenReturn(true);
+    doAnswer((invocation) -> {
+      throw new IOException();
+    }).when(reportFile).toPath();
+
+    importReport(reportFile);
+    assertThat(logTester.logs(LoggerLevel.WARN)).containsExactly(logMessage);
+    verify(mockAnalysisWarnings, times(1)).addWarning(logMessage);
+  }
+
+  @Test
+  void no_issues() {
+    File reportFile = new File("src/test/resources/hado-lint/emptyArray.json");
+    importReport(reportFile);
+    assertThat(context.allExternalIssues()).isEmpty();
+    verifyNoInteractions(mockAnalysisWarnings);
+  }
+
+  @Test
+  void invalid_issue() {
+    File reportFile = new File("src/test/resources/hado-lint/invalidIssue.json");
+    String logMessage = String.format("Hado-lint report importing: could not save 1 out of 1 issues from %s.", reportFile.getPath());
+    importReport(reportFile);
+    assertThat(context.allExternalIssues()).isEmpty();
+    assertThat(logTester.logs(LoggerLevel.WARN)).containsExactly(logMessage);
+    verify(mockAnalysisWarnings, times(1)).addWarning(logMessage);
+  }
+
+  @ParameterizedTest
+  @CsvSource({
+    "src/test/resources/hado-lint/jsonFormat/validIssue.json",
+    "src/test/resources/hado-lint/sonarqubeFormat/validIssue.json"})
+  void valid_issue(String reportPath) {
+    File reportFile = new File(reportPath);
+    importReport(reportFile);
+    assertThat(context.allExternalIssues()).hasSize(1);
+    ExternalIssue issue = context.allExternalIssues().iterator().next();
+    assertThat(issue.ruleId()).isEqualTo("DL3007");
+    assertThat(issue.type()).isEqualTo(RuleType.CODE_SMELL);
+    assertThat(issue.primaryLocation().message()).isEqualTo("Using latest is prone to errors if the image will ever update. Pin the version explicitly to a release tag");
+    AssertionsForClassTypes.assertThat(issue.primaryLocation().textRange().start().line()).isEqualTo(10);
```

Comment:
IMO in the whole class the simple `org.assertj.core.api.Assertions` can be used instead of `AssertionsForClassTypes` (and please only import static if possible)